app/services/Countries.scala

```diff
@@ -45,7 +45,13 @@ object Countries {
 
   val countryCodeMap: Map[String, Country] = countries.map(country => (country.countryCode, country)).toMap
 
-  def retrieveCountryNameFromCode(code: String): Country = countryCodeMap(code)
+  val countryNameMap: Map[String, Country] = countries.map(country => (country.countryName, country)).toMap
+
+  def retrieveCountryFromCode(code: String): Country = countryCodeMap(code)
```

Comment:
`findByCode`

Should probably also return an `Option[Country]` but that's up to you

Reply:
Changed